src/w3d/renderer/textureloader.cpp

```diff
@@ -344,32 +352,47 @@ void TextureLoader::Request_Foreground_Loading(TextureBaseClass *texture)
     TextureLoadTaskClass *thumb_task = texture->Get_Thumbnail_Load_Task();
 
     if (!TextureLoader::Is_DX8_Thread()) {
-        FastCriticalSectionClass::LockClass lock2(g_backgroundCritSec);
```

Comment:
Originally takes both locks at the same time. This can be risky and introduce ABA deadlock problem. This can be avoided by limiting scope.

Could perhaps deadlock with `LoaderThreadClass::Thread_Function`